contracts/eternal-storage/EternalStorageProxy.sol

```diff
@@ -1,48 +1,39 @@
 pragma solidity 0.5.2;
 
-import "./EternalStorage.sol";
+import "./OwnedEternalStorage.sol";
 import "../interfaces/IEternalStorageProxy.sol";
 
 
 /**
  * @title EternalStorageProxy
  * @dev This proxy holds the storage of the token contract and delegates every call to the current implementation set.
- * Besides, it allows to upgrade the token's behaviour towards further implementations, and provides
+ * It allows to upgrade the token's behaviour towards further implementations, and provides
  * authorization control functionalities
  */
-contract EternalStorageProxy is EternalStorage, IEternalStorageProxy {
-    bytes32 internal constant OWNER = keccak256("owner");
-
-    /**
-    * @dev This event will be emitted every time the implementation gets upgraded
-    * @param version representing the version number of the upgraded implementation
-    * @param implementation representing the address of the upgraded implementation
-    */
-    event Upgraded(uint256 version, address indexed implementation);
-    
-    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+contract EternalStorageProxy is IEternalStorageProxy, OwnedEternalStorage {
 
-    modifier onlyOwner() {
-        require(msg.sender == getOwner());
-        _;
-    }
+    /// @dev Emitted every time the implementation gets upgraded.
+    /// @param version The version number of the upgraded implementation.
+    /// @param implementation The address of the upgraded implementation.
+    event Upgraded(uint256 version, address indexed implementation);
 
-    constructor(address _implementationAddress, address _owner) public {
-        if (_implementationAddress != address(0)) {
-            require(_isContract(_implementationAddress));
-            _implementation = _implementationAddress;
-        }
-        if (_owner != address(0)) {
-            _setOwner(_owner);
+    /// @param implementation The address of the implementation.
+    /// @param owner The owner of the contract.  If set to `address(0)`, then
+    /// `msg.sender` will be used instead.
+    constructor(address implementation, address owner) public {
+        if (owner != address(0)) {
+            _owner = owner;
         } else {
-            _setOwner(msg.sender);
+            _owner = msg.sender;
         }
+
+        require(_isContract(implementation));
```

Comment:
These two statements should be inside the condition `if (_implementationAddress != address(0)) {` because we should have the possibility to deploy proxy contract without implementation which can be added later through `upgradeTo` function.